pkg/controller/main_test.go

```diff
@@ -147,13 +154,28 @@ func NewTestApplicationWithConfig(t *testing.T, configuration config.Configurati
 }
 
 func GivenIHaveToken(t *testing.T, e *httpexpect.Expect) string {
-	_, _, result := register(t, e)
-	require.Contains(t, result, "token", "result must contain token")
-	require.IsType(t, string(""), result["token"], "token must be a string")
-	return result["token"].(string)
+	email, password := register(t, e)
+	var loginRequest struct {
+		Email    string `json:"email"`
+		Password string `json:"password"`
+	}
+	loginRequest.Email = email
+	loginRequest.Password = password
+	var token string
+	{
+		response := e.POST("/api/authentication/login").
+			WithJSON(loginRequest).
+			Expect()
+
+		response.Status(http.StatusOK)
+		response.Cookie("M-Token").Value().NotEmpty()
+		token = response.Cookie("M-Token").Value().Raw()
```

Comment:
There is a const for the token name, this const should be used instead.

----

Maybe even make the name a configuration variable?